### ISSUE TITLE
Uyuni 4.0.2 - Salt bootstrap fails on CentOS7 with enabled EPEL (package conflict)

### ISSUE THREAD

**Issue by hunter86bg** [uyuni-project/uyuni#1275]
Hello All,

I have noticed that any RHEL 7/ CentOS 7 is failing to be bootstrapped or updated  when EPEL is enabled.
This behaviour was observed on our SUMA 3.2 , Uyuni 4.0.1 & Uyuni 4.0.2.

Based on the bootstrap script - all repos should be disabled before the packages are being installed but this is not true.
Bootstrap script:
```
[root@troubleshoot1 ~]# grep DISABLE_LOCAL_REPOS bootstrap.sh 
DISABLE_LOCAL_REPOS=1
```
When EPEL is enabled :
```
[root@troubleshoot1 ~]# ./bootstrap.sh 1-centos-7-x86_64
Uyuni Server Client bootstrap script v4.0


MINOR MANUAL EDITING OF THIS FILE MAY BE REQUIRED!

If this bootstrap script was created during the initial installation
of a Uyuni Server, the ACTIVATION_KEYS, and ORG_GPG_KEY values will
probably *not* be set (see below). If this is the case, please do the
following:
  - copy this file to a name specific to its use.
    (e.g., to bootstrap-SOME_NAME.sh - like bootstrap-web-servers.sh.)
  - on the website create an activation key or keys for the system(s) to
    be registered.
  - edit the values of the VARIABLES below (in this script) as
    appropriate:
    - ACTIVATION_KEYS needs to reflect the activation key(s) value(s)
      from the website. XKEY or XKEY,YKEY
      Please note that if you are using this script to boostrap minions,
      only the FIRST activation key will be used. Multiple activation keys
      are not supported with salt
    - ORG_GPG_KEY needs to be set to the name(s) of the corporate public
      GPG key filename(s) (residing in /srv/www/htdocs/pub) if appropriate. XKEY or XKEY,YKEY

Verify that the script variable settings are correct:
    - CLIENT_OVERRIDES should be only set differently if a customized
      client-config-overrides-VER.txt file was created with a different
      name.
    - ensure the value of HOSTNAME is correct.
    - ensure the value of ORG_CA_CERT is correct.

Enable this script: comment (with #'s) this block (or, at least just
the exit below, if present)


UPDATING RHN_REGISTER/UP2DATE CONFIGURATION FILES
-------------------------------------------------
* downloading necessary files
  client_config_update.py...
2019-08-10 23:30:45 URL:https://uyuni.localdomain.com/pub/bootstrap/client_config_update.py [6538/6538] -> "client_config_update.py" [1]
FINISHED --2019-08-10 23:30:45--
Total wall clock time: 0.03s
Downloaded: 1 files, 6.4K in 0s (105 MB/s)
  client-config-overrides.txt...
2019-08-10 23:30:45 URL:https://uyuni.localdomain.com/pub/bootstrap/client-config-overrides.txt [631/631] -> "client-config-overrides.txt" [1]
FINISHED --2019-08-10 23:30:45--
Total wall clock time: 0.02s
Downloaded: 1 files, 631 in 0s (56.9 MB/s)

PREPARE GPG KEYS AND CORPORATE PUBLIC CA CERT
-------------------------------------------------
* no organizational GPG keys to import

* attempting to install corporate public CA cert
2019-08-10 23:30:45 URL:https://uyuni.localdomain.com/pub/rhn-org-trusted-ssl-cert-1.0-1.noarch.rpm [13724/13724] -> "rhn-org-trusted-ssl-cert-1.0-1.noarch.rpm" [1]
FINISHED --2019-08-10 23:30:45--
Total wall clock time: 0.03s
Downloaded: 1 files, 13K in 0s (71.5 MB/s)
Preparing...                          ################################# [100%]
Updating / installing...
   1:rhn-org-trusted-ssl-cert-1.0-1   ################################# [100%]

CLEANING UP OLD SUSE MANAGER REPOSITORIES
-------------------------------------------------

CHECKING THE REGISTRATION STACK
-------------------------------------------------
* check for necessary packages being installed...
* client codebase is centos-7
package salt is not installed
package salt-minion is not installed
2019-08-10 23:30:46 URL:https://uyuni.localdomain.com/pub/repositories/centos/7/bootstrap/repodata/repomd.xml [2620/2620] -> "repomd.xml" [1]
FINISHED --2019-08-10 23:30:46--
Total wall clock time: 0.03s
Downloaded: 1 files, 2.6K in 0s (44.8 MB/s)
 adding client software repository at https://uyuni.localdomain.com/pub/repositories/centos/7/bootstrap
* going to install missing packages...
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.uni-sofia.bg
 * epel: mirrors.neterra.net
 * extras: centos.uni-sofia.bg
 * updates: centos.uni-sofia.bg
susemanager:bootstrap                                                                                                           | 2.6 kB  00:00:00     
susemanager:bootstrap/primary_db                                                                                                |  25 kB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python2-salt = 2019.2.0-16.2 for package: salt-2019.2.0-16.2.x86_64
---> Package salt-minion.x86_64 0:2019.2.0-16.2 will be installed
--> Running transaction check
---> Package python2-salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python-msgpack-python > 0.3 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-futures >= 2.0 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-pycrypto >= 2.6.1 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-pyzmq >= 2.2.0 for package: python2-salt-2019.2.0-16.2.x86_64
[epel]
---> Package python2-salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python-pyzmq >= 2.2.0 for package: python2-salt-2019.2.0-16.2.x86_64
Package python-zmq is obsoleted by python2-zmq, but obsoleting package does not provide for requirements
--> Running transaction check
---> Package python-backports.x86_64 0:1.0-8.el7 will be installed
---> Package python-ipaddress.noarch 0:1.0.16-2.el7 will be installed
---> Package python-six.noarch 0:1.9.0-2.el7 will be installed
---> Package python2-salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python-pyzmq >= 2.2.0 for package: python2-salt-2019.2.0-16.2.x86_64
Package python-zmq is obsoleted by python2-zmq, but obsoleting package does not provide for requirements
--> Finished Dependency Resolution
Error: Package: python2-salt-2019.2.0-16.2.x86_64 (susemanager:bootstrap)
           Requires: python-pyzmq >= 2.2.0
           Available: python-zmq-14.5.0-3.4.x86_64 (susemanager:bootstrap)
               python-pyzmq = 14.5.0-3.4
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
package salt is not installed
ERROR: Failed to install all missing packages.
``` 

When I manually disable EPEL on the system bootstrap succeeds:
```
[root@troubleshoot1 ~]# ./bootstrap.sh 1-centos-7-x86_64
Uyuni Server Client bootstrap script v4.0


MINOR MANUAL EDITING OF THIS FILE MAY BE REQUIRED!

If this bootstrap script was created during the initial installation
of a Uyuni Server, the ACTIVATION_KEYS, and ORG_GPG_KEY values will
probably *not* be set (see below). If this is the case, please do the
following:
  - copy this file to a name specific to its use.
    (e.g., to bootstrap-SOME_NAME.sh - like bootstrap-web-servers.sh.)
  - on the website create an activation key or keys for the system(s) to
    be registered.
  - edit the values of the VARIABLES below (in this script) as
    appropriate:
    - ACTIVATION_KEYS needs to reflect the activation key(s) value(s)
      from the website. XKEY or XKEY,YKEY
      Please note that if you are using this script to boostrap minions,
      only the FIRST activation key will be used. Multiple activation keys
      are not supported with salt
    - ORG_GPG_KEY needs to be set to the name(s) of the corporate public
      GPG key filename(s) (residing in /srv/www/htdocs/pub) if appropriate. XKEY or XKEY,YKEY

Verify that the script variable settings are correct:
    - CLIENT_OVERRIDES should be only set differently if a customized
      client-config-overrides-VER.txt file was created with a different
      name.
    - ensure the value of HOSTNAME is correct.
    - ensure the value of ORG_CA_CERT is correct.

Enable this script: comment (with #'s) this block (or, at least just
the exit below, if present)


UPDATING RHN_REGISTER/UP2DATE CONFIGURATION FILES
-------------------------------------------------
* downloading necessary files
  client_config_update.py...
2019-08-10 23:31:17 URL:https://uyuni.localdomain.com/pub/bootstrap/client_config_update.py [6538/6538] -> "client_config_update.py" [1]
FINISHED --2019-08-10 23:31:17--
Total wall clock time: 0.03s
Downloaded: 1 files, 6.4K in 0s (75.6 MB/s)
  client-config-overrides.txt...
2019-08-10 23:31:17 URL:https://uyuni.localdomain.com/pub/bootstrap/client-config-overrides.txt [631/631] -> "client-config-overrides.txt" [1]
FINISHED --2019-08-10 23:31:17--
Total wall clock time: 0.03s
Downloaded: 1 files, 631 in 0s (72.3 MB/s)

PREPARE GPG KEYS AND CORPORATE PUBLIC CA CERT
-------------------------------------------------
* no organizational GPG keys to import

* attempting to install corporate public CA cert
2019-08-10 23:31:17 URL:https://uyuni.localdomain.com/pub/rhn-org-trusted-ssl-cert-1.0-1.noarch.rpm [13724/13724] -> "rhn-org-trusted-ssl-cert-1.0-1.noarch.rpm" [1]
FINISHED --2019-08-10 23:31:17--
Total wall clock time: 0.03s
Downloaded: 1 files, 13K in 0s (220 MB/s)
Preparing...                          ################################# [100%]
Updating / installing...
   1:rhn-org-trusted-ssl-cert-1.0-1   ################################# [100%]

CLEANING UP OLD SUSE MANAGER REPOSITORIES
-------------------------------------------------

CHECKING THE REGISTRATION STACK
-------------------------------------------------
* check for necessary packages being installed...
* client codebase is centos-7
package salt is not installed
package salt-minion is not installed
2019-08-10 23:31:18 URL:https://uyuni.localdomain.com/pub/repositories/centos/7/bootstrap/repodata/repomd.xml [2620/2620] -> "repomd.xml" [1]
FINISHED --2019-08-10 23:31:18--
Total wall clock time: 0.02s
Downloaded: 1 files, 2.6K in 0s (52.4 MB/s)
 adding client software repository at https://uyuni.localdomain.com/pub/repositories/centos/7/bootstrap
* going to install missing packages...
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.uni-sofia.bg
 * extras: centos.uni-sofia.bg
 * updates: centos.uni-sofia.bg
susemanager:bootstrap                                                                                                           | 2.6 kB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python2-salt = 2019.2.0-16.2 for package: salt-2019.2.0-16.2.x86_64
---> Package salt-minion.x86_64 0:2019.2.0-16.2 will be installed
--> Running transaction check
---> Package python2-salt.x86_64 0:2019.2.0-16.2 will be installed
--> Processing Dependency: python-msgpack-python > 0.3 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-futures >= 2.0 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-pycrypto >= 2.6.1 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-pyzmq >= 2.2.0 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-requests >= 1.0.0 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-tornado >= 4.2.1 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-jinja2 for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-markupsafe for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-psutil for package: python2-salt-2019.2.0-16.2.x86_64
--> Processing Dependency: python-yaml for package: python2-salt-2019.2.0-16.2.x86_64
--> Running transaction check
---> Package PyYAML.x86_64 0:3.10-11.el7 will be installed
--> Processing Dependency: libyaml-0.so.2()(64bit) for package: PyYAML-3.10-11.el7.x86_64
---> Package python-jinja2.noarch 0:2.7.2-3.el7_6 will be installed
--> Processing Dependency: python-babel >= 0.8 for package: python-jinja2-2.7.2-3.el7_6.noarch
---> Package python-markupsafe.x86_64 0:0.11-10.el7 will be installed
---> Package python-msgpack-python.x86_64 0:0.4.6-2.4 will be installed
---> Package python-psutil.x86_64 0:1.2.1-0.10.4 will be installed
---> Package python-pycrypto.x86_64 0:2.6.1-2.4 will be installed
---> Package python-requests.noarch 0:2.6.0-1.el7_1 will be installed
--> Processing Dependency: python-urllib3 >= 1.10.2-1 for package: python-requests-2.6.0-1.el7_1.noarch
--> Processing Dependency: python-chardet >= 2.2.1-1 for package: python-requests-2.6.0-1.el7_1.noarch
---> Package python-tornado.x86_64 0:4.2.1-4.el7 will be installed
--> Processing Dependency: python-backports-ssl_match_hostname for package: python-tornado-4.2.1-4.el7.x86_64
---> Package python-zmq.x86_64 0:14.5.0-3.4 will be installed
--> Processing Dependency: libzmq.so.4()(64bit) for package: python-zmq-14.5.0-3.4.x86_64
---> Package python2-futures.noarch 0:3.1.1-5.el7 will be installed
--> Running transaction check
---> Package libyaml.x86_64 0:0.1.4-11.el7_0 will be installed
---> Package python-babel.noarch 0:0.9.6-8.el7 will be installed
---> Package python-backports-ssl_match_hostname.noarch 0:3.5.0.1-1.el7 will be installed
--> Processing Dependency: python-ipaddress for package: python-backports-ssl_match_hostname-3.5.0.1-1.el7.noarch
--> Processing Dependency: python-backports for package: python-backports-ssl_match_hostname-3.5.0.1-1.el7.noarch
---> Package python-chardet.noarch 0:2.2.1-1.el7_1 will be installed
---> Package python-urllib3.noarch 0:1.10.2-5.el7 will be installed
--> Processing Dependency: python-six for package: python-urllib3-1.10.2-5.el7.noarch
---> Package zeromq.x86_64 0:4.0.5-1.4 will be installed
--> Running transaction check
---> Package python-backports.x86_64 0:1.0-8.el7 will be installed
---> Package python-ipaddress.noarch 0:1.0.16-2.el7 will be installed
---> Package python-six.noarch 0:1.9.0-2.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=======================================================================================================================================================
 Package                                             Arch                   Version                        Repository                             Size
=======================================================================================================================================================
Installing:
 salt                                                x86_64                 2019.2.0-16.2                  susemanager:bootstrap                 153 k
 salt-minion                                         x86_64                 2019.2.0-16.2                  susemanager:bootstrap                 151 k
Installing for dependencies:
 PyYAML                                              x86_64                 3.10-11.el7                    base                                  153 k
 libyaml                                             x86_64                 0.1.4-11.el7_0                 base                                   55 k
 python-babel                                        noarch                 0.9.6-8.el7                    base                                  1.4 M
 python-backports                                    x86_64                 1.0-8.el7                      base                                  5.8 k
 python-backports-ssl_match_hostname                 noarch                 3.5.0.1-1.el7                  base                                   13 k
 python-chardet                                      noarch                 2.2.1-1.el7_1                  base                                  227 k
 python-ipaddress                                    noarch                 1.0.16-2.el7                   base                                   34 k
 python-jinja2                                       noarch                 2.7.2-3.el7_6                  susemanager:bootstrap                 518 k
 python-markupsafe                                   x86_64                 0.11-10.el7                    base                                   25 k
 python-msgpack-python                               x86_64                 0.4.6-2.4                      susemanager:bootstrap                  75 k
 python-psutil                                       x86_64                 1.2.1-0.10.4                   susemanager:bootstrap                 114 k
 python-pycrypto                                     x86_64                 2.6.1-2.4                      susemanager:bootstrap                 454 k
 python-requests                                     noarch                 2.6.0-1.el7_1                  base                                   94 k
 python-six                                          noarch                 1.9.0-2.el7                    base                                   29 k
 python-tornado                                      x86_64                 4.2.1-4.el7                    base                                  641 k
 python-urllib3                                      noarch                 1.10.2-5.el7                   base                                  102 k
 python-zmq                                          x86_64                 14.5.0-3.4                     susemanager:bootstrap                 492 k
 python2-futures                                     noarch                 3.1.1-5.el7                    base                                   29 k
 python2-salt                                        x86_64                 2019.2.0-16.2                  susemanager:bootstrap                 9.6 M
 zeromq                                              x86_64                 4.0.5-1.4                      susemanager:bootstrap                 582 k

Transaction Summary
=======================================================================================================================================================
Install  2 Packages (+20 Dependent packages)

Total download size: 15 M
Installed size: 66 M
Downloading packages:
(1/22): PyYAML-3.10-11.el7.x86_64.rpm                                                                                           | 153 kB  00:00:00     
(2/22): libyaml-0.1.4-11.el7_0.x86_64.rpm                                                                                       |  55 kB  00:00:00     
(3/22): python-backports-1.0-8.el7.x86_64.rpm                                                                                   | 5.8 kB  00:00:00     
(4/22): python-backports-ssl_match_hostname-3.5.0.1-1.el7.noarch.rpm                                                            |  13 kB  00:00:00     
(5/22): python-chardet-2.2.1-1.el7_1.noarch.rpm                                                                                 | 227 kB  00:00:00     
(6/22): python-ipaddress-1.0.16-2.el7.noarch.rpm                                                                                |  34 kB  00:00:00     
(7/22): python-markupsafe-0.11-10.el7.x86_64.rpm                                                                                |  25 kB  00:00:00     
(8/22): python-babel-0.9.6-8.el7.noarch.rpm                                                                                     | 1.4 MB  00:00:00     
(9/22): python-jinja2-2.7.2-3.el7_6.noarch.rpm                                                                                  | 518 kB  00:00:00     
(10/22): python-psutil-1.2.1-0.10.4.x86_64.rpm                                                                                  | 114 kB  00:00:00     
(11/22): python-msgpack-python-0.4.6-2.4.x86_64.rpm                                                                             |  75 kB  00:00:00     
(12/22): python-pycrypto-2.6.1-2.4.x86_64.rpm                                                                                   | 454 kB  00:00:00     
(13/22): python-six-1.9.0-2.el7.noarch.rpm                                                                                      |  29 kB  00:00:00     
(14/22): python-requests-2.6.0-1.el7_1.noarch.rpm                                                                               |  94 kB  00:00:00     
(15/22): python2-futures-3.1.1-5.el7.noarch.rpm                                                                                 |  29 kB  00:00:00     
(16/22): python-zmq-14.5.0-3.4.x86_64.rpm                                                                                       | 492 kB  00:00:00     
(17/22): salt-2019.2.0-16.2.x86_64.rpm                                                                                          | 153 kB  00:00:00     
(18/22): python-urllib3-1.10.2-5.el7.noarch.rpm                                                                                 | 102 kB  00:00:00     
(19/22): salt-minion-2019.2.0-16.2.x86_64.rpm                                                                                   | 151 kB  00:00:00     
(20/22): zeromq-4.0.5-1.4.x86_64.rpm                                                                                            | 582 kB  00:00:00     
(21/22): python-tornado-4.2.1-4.el7.x86_64.rpm                                                                                  | 641 kB  00:00:00     
(22/22): python2-salt-2019.2.0-16.2.x86_64.rpm                                                                                  | 9.6 MB  00:00:00     
-------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                   14 MB/s |  15 MB  00:00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
Warning: RPMDB altered outside of yum.
  Installing : python-ipaddress-1.0.16-2.el7.noarch                                                                                               1/22 
  Installing : python-markupsafe-0.11-10.el7.x86_64                                                                                               2/22 
  Installing : python-chardet-2.2.1-1.el7_1.noarch                                                                                                3/22 
  Installing : libyaml-0.1.4-11.el7_0.x86_64                                                                                                      4/22 
  Installing : PyYAML-3.10-11.el7.x86_64                                                                                                          5/22 
  Installing : python-pycrypto-2.6.1-2.4.x86_64                                                                                                   6/22 
  Installing : zeromq-4.0.5-1.4.x86_64                                                                                                            7/22 
  Installing : python-zmq-14.5.0-3.4.x86_64                                                                                                       8/22 
  Installing : python-backports-1.0-8.el7.x86_64                                                                                                  9/22 
  Installing : python-backports-ssl_match_hostname-3.5.0.1-1.el7.noarch                                                                          10/22 
  Installing : python-tornado-4.2.1-4.el7.x86_64                                                                                                 11/22 
  Installing : python-babel-0.9.6-8.el7.noarch                                                                                                   12/22 
  Installing : python-jinja2-2.7.2-3.el7_6.noarch                                                                                                13/22 
  Installing : python-psutil-1.2.1-0.10.4.x86_64                                                                                                 14/22 
  Installing : python-six-1.9.0-2.el7.noarch                                                                                                     15/22 
  Installing : python-urllib3-1.10.2-5.el7.noarch                                                                                                16/22 
  Installing : python-requests-2.6.0-1.el7_1.noarch                                                                                              17/22 
  Installing : python-msgpack-python-0.4.6-2.4.x86_64                                                                                            18/22 
  Installing : python2-futures-3.1.1-5.el7.noarch                                                                                                19/22 
  Installing : salt-2019.2.0-16.2.x86_64                                                                                                         20/22 
  Installing : python2-salt-2019.2.0-16.2.x86_64                                                                                                 21/22 
  Installing : salt-minion-2019.2.0-16.2.x86_64                                                                                                  22/22 
  Verifying  : python-backports-ssl_match_hostname-3.5.0.1-1.el7.noarch                                                                           1/22 
  Verifying  : python2-futures-3.1.1-5.el7.noarch                                                                                                 2/22 
  Verifying  : python-zmq-14.5.0-3.4.x86_64                                                                                                       3/22 
  Verifying  : salt-minion-2019.2.0-16.2.x86_64                                                                                                   4/22 
  Verifying  : python2-salt-2019.2.0-16.2.x86_64                                                                                                  5/22 
  Verifying  : python-msgpack-python-0.4.6-2.4.x86_64                                                                                             6/22 
  Verifying  : python-jinja2-2.7.2-3.el7_6.noarch                                                                                                 7/22 
  Verifying  : python-six-1.9.0-2.el7.noarch                                                                                                      8/22 
  Verifying  : python-urllib3-1.10.2-5.el7.noarch                                                                                                 9/22 
  Verifying  : python-tornado-4.2.1-4.el7.x86_64                                                                                                 10/22 
  Verifying  : python-markupsafe-0.11-10.el7.x86_64                                                                                              11/22 
  Verifying  : python-psutil-1.2.1-0.10.4.x86_64                                                                                                 12/22 
  Verifying  : python-babel-0.9.6-8.el7.noarch                                                                                                   13/22 
  Verifying  : python-backports-1.0-8.el7.x86_64                                                                                                 14/22 
  Verifying  : zeromq-4.0.5-1.4.x86_64                                                                                                           15/22 
  Verifying  : python-pycrypto-2.6.1-2.4.x86_64                                                                                                  16/22 
  Verifying  : libyaml-0.1.4-11.el7_0.x86_64                                                                                                     17/22 
  Verifying  : salt-2019.2.0-16.2.x86_64                                                                                                         18/22 
  Verifying  : python-ipaddress-1.0.16-2.el7.noarch                                                                                              19/22 
  Verifying  : python-chardet-2.2.1-1.el7_1.noarch                                                                                               20/22 
  Verifying  : PyYAML-3.10-11.el7.x86_64                                                                                                         21/22 
  Verifying  : python-requests-2.6.0-1.el7_1.noarch                                                                                              22/22 

Installed:
  salt.x86_64 0:2019.2.0-16.2                                            salt-minion.x86_64 0:2019.2.0-16.2                                           

Dependency Installed:
  PyYAML.x86_64 0:3.10-11.el7                 libyaml.x86_64 0:0.1.4-11.el7_0                               python-babel.noarch 0:0.9.6-8.el7        
  python-backports.x86_64 0:1.0-8.el7         python-backports-ssl_match_hostname.noarch 0:3.5.0.1-1.el7    python-chardet.noarch 0:2.2.1-1.el7_1    
  python-ipaddress.noarch 0:1.0.16-2.el7      python-jinja2.noarch 0:2.7.2-3.el7_6                          python-markupsafe.x86_64 0:0.11-10.el7   
  python-msgpack-python.x86_64 0:0.4.6-2.4    python-psutil.x86_64 0:1.2.1-0.10.4                           python-pycrypto.x86_64 0:2.6.1-2.4       
  python-requests.noarch 0:2.6.0-1.el7_1      python-six.noarch 0:1.9.0-2.el7                               python-tornado.x86_64 0:4.2.1-4.el7      
  python-urllib3.noarch 0:1.10.2-5.el7        python-zmq.x86_64 0:14.5.0-3.4                                python2-futures.noarch 0:3.1.1-5.el7     
  python2-salt.x86_64 0:2019.2.0-16.2         zeromq.x86_64 0:4.0.5-1.4                                    

Complete!
salt-2019.2.0-16.2.x86_64
salt-minion-2019.2.0-16.2.x86_64
Loaded plugins: fastestmirror, yumnotify
Loading mirror speeds from cached hostfile
 * base: centos.uni-sofia.bg
 * extras: centos.uni-sofia.bg
 * updates: centos.uni-sofia.bg
No packages marked for update

REGISTRATION
------------
* registering
* removing TLS certificate used for bootstrap
  (will be re-added via salt state)
* starting salt daemon and enabling it during boot
Created symlink from /etc/systemd/system/multi-user.target.wants/salt-minion.service to /usr/lib/systemd/system/salt-minion.service.
-bootstrap complete-
```

Why does 'DISABLE_LOCAL_REPOS=1' doesn't work ?
Anyone can reproduce this behaviour ?

**Comment by hunter86bg** [comment]
Using the following bypasses any local repos: 
```
yum  install --disablerepo='*' --enablerepo='susemanager:bootstrap' salt salt-minion
```

So the bootstrap script could be changed like this:
```
if [ -z "$Y_MISSING" ]; then
        echo "  no packages missing."
    else
        echo "* going to install missing packages..."

       if [ "$DISABLE_LOCAL_REPOS" -eq "1" ]
           then 
                  yum -y install --disablerepo='*' --enablerepo='susemanager:bootstrap'  $Y_MISSING
           else
                  yum -y install $Y_MISSING
       fi

        for P in $Y_MISSING; do
            rpm -q "$P" || {
            echo "ERROR: Failed to install all missing packages."
            exit 1
        }
        done
    fi

**Comment by hunter86bg** [comment]
It seems that the bootstrap ends successfully , but the salt-minion fails to start. I'm not sure if the issue is that all repos were disabled during salt installation or due to multiple re-registrations.

**Comment by hunter86bg** [comment]
Reverted both CentOS7 VMs from snapshot, changed /etc/machine-id to another value.
With the modification above I have managed to bootstrap successfully and later - update the VMs.

**Comment by hunter86bg** [comment]
Another approach that comes to my mind is to set lower priority for epel repo, which will help during updates.

**Comment by paususe** [comment]
Does this still happen with Uyuni 2020.01 and the updated python2-zmq packages from EPEL?

**Comment by brejoc** [comment]
This should be fixed now: https://bugzilla.redhat.com/show_bug.cgi?id=1789889

I'm closing this. @hunter86bg, please re-open this issue if you still run into problems after the fixed build was pushed to EPEL.
